### PR TITLE
Add ability to remove credential

### PR DESCRIPTION
**Pull request by amanji**
# Summary of Changes

* Credentials can be removed from the details screen
* Adds new common delete modal (to be re-used with deleting contacts)
* Updates Bifold with latest react-hooks

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [ ] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [ ] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [ ] Updated documentation as needed for changed code and new or modified features;
- [ ] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: core/App/components/modals/CommonDeleteModal.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import { FlatList } from 'react-native-gesture-handler'
+import Icon from 'react-native-vector-icons/MaterialIcons'
+
+import Button, { ButtonType } from '../../components/buttons/Button'
+import FauxNavigationBar from '../../components/views/FauxNavigationBar'
+import { useTheme } from '../../contexts/theme'
+import { testIdWithKey } from '../../utils/testable'
+
+interface CommonDeleteModalProps {
+  onSubmit?: () => void
```
Comment (jleach) [inline core/App/components/modals/CommonDeleteModal.tsx]: I think we have a type for these: `GenericFn` might be it if I remember.

**Review by jleach — DISMISSED** [review]

**Comment by jleach** [comment]
@PatStLouis Able to supply the accompanying French translation for this feature?

**Comment by PatStLouis** [comment]
@jleach we will have a look, is it possible to add the fr localization file with placeholders in the PR like last time?

**Comment by amanji** [comment]
> @jleach we will have a look, is it possible to add the fr localization file with placeholders in the PR like last time?

Sure thing, I can commit an update with that

**Review by jleach — APPROVED** [review]

**Comment by amanji** [comment]
https://github.com/hyperledger/aries-mobile-agent-react-native/issues/385 

Added as a reminder

**Pull request merged**